docs/pages/guides/authentication.mdx

```diff
@@ -770,6 +771,137 @@ function useMounted() {
+
+<Tabs tabs={["Auth Code"]}>
+
+<Tab>
+
+{/* prettier-ignore */}
+```tsx
+import * as React from "react";
+import * as WebBrowser from "expo-web-browser";
+import * as AuthSession from "expo-auth-session";
+import { Button, Alert, Text, View } from "react-native";
+
+WebBrowser.maybeCompleteAuthSession();
+
+// Replace this with your Descope Project ID
+const descopeProjectId = "<Descope Project ID>";
+const descopeUrl = `https://api.descope.com/${descopeProjectId}`;
+const redirectUri = AuthSession.makeRedirectUri();
+
+export default function App() {
+  const [authTokens, setAuthTokens] = React.useState(null);
+  const discovery = AuthSession.useAutoDiscovery(descopeUrl);
+
+  const [request, response, promptAsync] = AuthSession.useAuthRequest(
+    {
+      clientId: descopeProjectId,
+      responseType: AuthSession.ResponseType.Code,
+      redirectUri,
+      usePKCE: true,
+      scopes: ["openid", "profile", "email"],
+    },
+    discovery
+  );
+
+  React.useEffect(() => {
+    /* @info Exchange the code for an access token in a server. */
+    const exchangeFn = async (exchangeTokenReq) => {
+    /* @end */
+      try {
+        const exchangeTokenResponse = await AuthSession.exchangeCodeAsync(
+          exchangeTokenReq,
+          discovery
+        );
+        setAuthTokens(exchangeTokenResponse);
+      } catch (error) {
+        console.error(error);
+      }
+    };
+    if (response) {
+      if (response.error) {
+        Alert.alert(
+          "Authentication error",
+          response.params.error_description || "something went wrong"
+        );
+        return;
+      }
+      if (response.type === "success") {
+        exchangeFn({
+          clientId: descopeProjectId,
+          code: response.params.code,
+          redirectUri,
+          extraParams: {
+            code_verifier: request.codeVerifier,
+          },
+        });
+      }
+    }
+  }, [discovery, request, response]);
+
+  const logout = async () => {
+    const revokeResponse = await AuthSession.revokeAsync(
+      {
+        clientId: descopeProjectId,
+        token: authTokens.refreshToken,
+      },
+      discovery
+    );
+    if (revokeResponse) {
+      setAuthTokens(null);
+    }
+  };
+  console.log("authTokens: " + JSON.stringify(authTokens));
+  return (
+    <View
+      style={{
+        flex: 1,
+        justifyContent: "center",
+        alignItems: "center",
+        padding: 20,
+      }}
+    >
+      {authTokens ? (
+        <Button title="Logout" onPress={() => logout()} />
+      ) : (
+        <Button
+          /* @info Disable the button until the request is loaded asynchronously. */
+          disabled={!request}
+          /* @end */
+          title="Login"
+          /* @info Prompt the user to authenticate in a user interaction or web browsers will block it. */
+          onPress={() => promptAsync()}
+          /* @end */
+        />
+      )}
+    </View>
+  );
+}
```

Comment:
```suggestion
import * as React from 'react';
import * as WebBrowser from 'expo-web-browser';
import * as AuthSession from 'expo-auth-session';
import { Button, View } from 'react-native';

WebBrowser.maybeCompleteAuthSession();

const descopeProjectId = '<Descope Project ID>'; // Replace this with your Descope Project ID
const descopeUrl = `https://api.descope.com/${descopeProjectId}`;
const redirectUri = AuthSession.makeRedirectUri();

export default function App() {
  const [authTokens, setAuthTokens] = React.useState(null);
  const discovery = AuthSession.useAutoDiscovery(descopeUrl);

  const [request, response, promptAsync] = AuthSession.useAuthRequest(
    {
      clientId: descopeProjectId,
      responseType: AuthSession.ResponseType.Code,
      redirectUri,
      usePKCE: true,
      scopes: ['openid', 'profile', 'email'],
    },
    discovery
  );

  React.useEffect(() => {
    if (response) {
      if (response.error) {
        console.error(
          'Authentication error',
          response.params.error_description || 'something went wrong'
        );
        return;
      }
      if (response.type === 'success') {
        /* @info Exchange the code for an access token in a server. */
        const exchangeFn = async (exchangeTokenReq) => {
        /* @end */
          try {
            const exchangeTokenResponse = await AuthSession.exchangeCodeAsync(
              exchangeTokenReq,
              discovery
            );
            setAuthTokens(exchangeTokenResponse);
          } catch (error) {
            console.error(error);
          }
        };

        exchangeFn({
          clientId: descopeProjectId,
          code: response.params.code,
          redirectUri,
          extraParams: {
            code_verifier: request.codeVerifier,
          },
        });
      }
    }
  }, [discovery, request, response]);

  const logout = async () => {
    const revokeResponse = await AuthSession.revokeAsync(
      {
        clientId: descopeProjectId,
        token: authTokens.refreshToken,
      },
      discovery
    );
    if (revokeResponse) {
      setAuthTokens(null);
    }
  };

  return (
    <View>
      {authTokens ? (
        <Button title="Logout" onPress={logout} />
      ) : (
        <Button
          /* @info Disable the button until the request is loaded asynchronously. */
          disabled={!request}
          /* @end */
          title="Login"
          /* @info Prompt the user to authenticate in a user interaction or web browsers will block it. */
          onPress={promptAsync}
          /* @end */
        />
      )}
    </View>
  );
}
```

Let's concise the example a bit.

Reply:
Would it be possible to keep the code exchange part for the token, so customers can copy the whole thing, plug it in, and have it work right away without them having to implement that themselves? The rest is fine to remove.

Reply:
The exchange function is still in the code, but it has been moved down to the "success" result code branch, where it's called, there is no need to initialize it earlier.